test/lsp/lsp_rename.ml

```diff
@@ -0,0 +1,352 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                        SuperBOL OSS Studio                             *)
+(*                                                                        *)
+(*  Copyright (c) 2022-2023 OCamlPro SAS                                  *)
+(*                                                                        *)
+(* All rights reserved.                                                   *)
+(* This source code is licensed under the GNU Affero General Public       *)
+(* License version 3 found in the LICENSE.md file in the root directory   *)
+(* of this source tree.                                                   *)
+(*                                                                        *)
+(**************************************************************************)
+
+open EzCompat                                                    (* StringMap *)
+open Lsp.Types
+open Lsp_testing
+
+let pp_assoc_elem ppf ((uri, edits): DocumentUri.t * TextEdit.t list) =
+  let pp_range ppf range =
+  let location_as_srcloc = new srcloc_resuscitator_cache in
+  location_as_srcloc#pp ppf @@ Location.create ~range ~uri
+  in
+  let pp_edit ppf (edit: TextEdit.t) =
+    Fmt.pf ppf "%s at " edit.newText;
+    pp_range ppf edit.range
+  in
+  Fmt.(list pp_edit) ppf edits
+
+let count l =
+  List.fold_left begin fun acc (_, t) -> acc + List.length t end 0 l
+
+let rename_positions ?(ignore_when_copybook=false) ?(copybooks=[]) (doc, positions) : string -> unit =
+  let { end_with_postproc; projdir }, server = make_lsp_project () in
+  let server = List.fold_left begin fun server (name, document) ->
+      add_cobol_doc server ~projdir name document
+      |> fst
+    end server copybooks in
+  let server, prog = add_cobol_doc server ~projdir "prog.cob" doc in
+  let rename_at_position ?key (position: Position.t) =
+    let params = RenameParams.create ~newName:"aNewName" ~position ~textDocument:prog () in
+    Pretty.out "%a(line %d, character %d):\n"
+      Fmt.(option ~none:nop (string ++ sp)) key
+      position.line position.character;
+    begin
+      try
+      match LSP.Request.rename ~ignore_when_copybook server params with
+      | { changes = None; _ } ->
+        Pretty.out "No renames@."
+      | { changes = Some assoc; _ } ->
+        Pretty.out "@.@[<hv 4>%d rename entries:@;%a@]@\n"
+          (count assoc)
+          (Fmt.list ~sep:Fmt.sp pp_assoc_elem) assoc
+      with _ -> Pretty.out "Failed rename@."
```

Comment:
I'd rather have the expected exception(s) listed here for the test to be more precise.

Reply:
Even better if the try block is unnecessary ;-)